security/authentication.md

```diff
@@ -0,0 +1,721 @@
+# Authentication Patterns
+
+## Overview
+
+Comprehensive authentication patterns for userver applications, covering JWT token authentication, API key authentication, OAuth2 integration, session management, and multi-factor authentication patterns.
+
+## Core Authentication Components
+
+### JWT Authentication
+- [`crypto::jwt::Jwt`](https://userver.tech/d7/d32/classcrypto_1_1jwt_1_1Jwt.html) - JWT token handling and validation
+- [`server::handlers::auth::AuthCheckerBase`](https://userver.tech/d6/db7/classserver_1_1handlers_1_1auth_1_1AuthCheckerBase.html) - Base class for custom authentication checkers
+- [`crypto::algorithm::StringsEqualConstTimeComparator`](https://userver.tech/de/d55/structcrypto_1_1algorithm_1_1StringsEqualConstTimeComparator.html) - Timing attack prevention for token comparison
+
+### API Key Authentication
+- [`components::PostgreCache`](https://userver.tech/d2/d8d/classcomponents_1_1PostgreCache.html) - Caching authentication data
+- [`server::auth::UserAuthInfo::Ticket`](https://userver.tech/d4/d67/classutils_1_1StrongTypedef.html) - Strongly typed authentication tickets
+- [`crypto::algorithm`](https://userver.tech/de/d55/algorithm_8hpp.html) - Cryptographic algorithms for secure comparisons
+
+### OAuth2 Integration
+- [`clients::http::Client`](https://userver.tech/d5/dee/classclients_1_1http_1_1Client.html) - HTTP client for OAuth2 provider communication
+- [`formats::json`](https://userver.tech/d2/d20/md_en_2userver_2formats.html) - JSON parsing for OAuth2 responses
+- [`server::http::HttpRequest`](https://userver.tech/d3/d44/classserver_1_1http_1_1HttpRequest.html) - HTTP request handling for OAuth2 flows
+
+## JWT Authentication Implementation
+
+### JWT Token Structure and Validation
+```cpp
```

Comment:
all examples in docs will become outdated very soon
please move them to samples and reference them from docs